lib/trino-hive-formats/src/main/java/io/trino/hive/formats/esri/EsriDeserializer.java

```diff
@@ -0,0 +1,371 @@
+            }
+            else {
+                throw new RuntimeException("Column '" + columnName + "' with type: " + columnType.getDisplayName() + " is not supported");
+            }
+        }
+        catch (Exception e) {
+            // invalid columns are ignored
+            if (nullOnParseError) {
+                builder.appendNull();
+            }
+            else {
+                throw new TrinoException(GENERIC_USER_ERROR, "Error Parsing a column in the table: " + e.getMessage(), e);
+            }
+        }
+    }
+
+    private static DecodedTimestamp createDecodedTimestamp(Timestamp timestamp)
+    {
+        long millis = timestamp.getTime();
+        long epochSeconds = floorDiv(millis, (long) MILLISECONDS_PER_SECOND);
+        long fractionalSecond = floorMod(millis, (long) MILLISECONDS_PER_SECOND);
+        int nanosOfSecond = toIntExact(fractionalSecond * (long) NANOSECONDS_PER_MILLISECOND);
+
+        return new DecodedTimestamp(epochSeconds, nanosOfSecond);
+    }
+
+    private static void serializeDecimal(String value, DecimalType decimalType, BlockBuilder builder)
+    {
+        BigDecimal bigDecimal;
+        try {
+            bigDecimal = new BigDecimal(value).setScale(DecimalConversions.intScale(decimalType.getScale()), HALF_UP);
+        }
+        catch (NumberFormatException e) {
+            throw new NumberFormatException(format("Cannot convert '%s' to %s. Value is not a number.", value, decimalType));
+        }
+
+        if (overflows(bigDecimal, decimalType.getPrecision())) {
+            throw new IllegalArgumentException(format("Cannot convert '%s' to %s. Value too large.", value, decimalType));
+        }
+
+        if (decimalType.isShort()) {
+            decimalType.writeLong(builder, bigDecimal.unscaledValue().longValueExact());
+        }
+        else {
+            decimalType.writeObject(builder, Int128.valueOf(bigDecimal.unscaledValue()));
+        }
+    }
+
+    private Date parseDate(JsonParser parser)
+            throws IOException
+    {
+        Date jsd = null;
+        if (JsonToken.VALUE_NUMBER_INT.equals(parser.getCurrentToken())) {
+            long epoch = parser.getLongValue();
+            jsd = new Date(epoch - (long) tz.getOffset(epoch));
+        }
+        else {
+            try {
+                long epoch = parseTime(parser.getText(), "yyyy-MM-dd");
+                jsd = new Date(epoch + 43200000L);
+            }
+            catch (ParseException e) {
+            }
+        }
+
+        return jsd;
+    }
+
+    private Timestamp parseTime(JsonParser parser)
+            throws IOException
+    {
+        Timestamp jst = null;
+        if (JsonToken.VALUE_NUMBER_INT.equals(parser.getCurrentToken())) {
+            long epoch = parser.getLongValue();
+            jst = new Timestamp(epoch);
+        }
+        else {
+            String value = parser.getText();
+            int point = value.indexOf(46);
```

Comment:
Looks like this came from a decompiler, this is equivalent to `value.indexOf('.')`